var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -207,6 +207,10 @@ def setup_dependent_package(self, module, dep_spec):
         self.spec.mpicxx = join_path(self.prefix.bin, 'mpicxx')
         self.spec.mpifc  = join_path(self.prefix.bin, 'mpif90')
         self.spec.mpif77 = join_path(self.prefix.bin, 'mpif77')
+        if '+slurm' in self.spec:
+            self.spec.run = join_path(self.prefix.bin, 'srun')
```

Comment:
are you sure it's external? There is no `depends_on('slurm')`, it's only a variant. Can you check where `srun` on your system?